docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -0,0 +1,378 @@
+
+You can specify the Beat to deploy and its version through `type` and `version` elements. ECK computes the correct path to the container image, so the `image` element doesn't have to be provided. ECK creates a new user in Elasticsearch with a minimal set of appropriate roles and permissions to enable the use of all Beats features.
+
+[id="{p}-beat-deploy-community-beat"]
+=== Deploy a Community Beat
+
+ECK supports the deployment of any Community Beat. `type` and `version` specification elements have to be provided. In addition:
+
+- `image` must point to the image to be deployed
+-  a role must exist in Elasticsearch and have the permissions required by the Beat. The role name has to be `eck_beat_type_role`, where `type` is the Beat type. For example, when deploying `type: kafkabeat`, the role name would be `eck_beat_kafkabeat_role`.
+
+
+[id="{p}-beat-set-up-kibana-dashboards"]
+=== Set up Kibana dashboards
+
+ECK can instruct Beats to set up example dashboards packaged with the Beat. To enable this, set the `kibanaRef` element in the specification to point to ECK-managed Kibana deployment:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: quickstart
+spec:
+  kibanaRef:
+    name: quickstart
+...
+----
+
+ECK will create a new user in Elasticsearch with a minimal set of appropriate roles and permissions that is needed for dashboard setup.
+
+
+[id="{p}-beat-secrets-keystore-for-secure-settings"]
+=== Secrets keystore for secure settings
+
+Beats offer a secret keystore for sensitive settings that need to be provided in the config, like for example passwords. This avoids storing them in the config directly.
+
+ECK exposes that mechanism with the `secureSettings` element in the specification. <<{p}-es-secure-settings,Similar to Elasticsearch>>, you can use Kubernetes Secrets to provide the settings in a secure way:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: quickstart
+spec:
+  secureSettings:
+  - secretName: agent-name-secret
+  config:
+    name: ${AGENT_NAME_VAR}
+...
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: agent-name-secret
+stringData:
+  AGENT_NAME_VAR: id_007
+----
+
+See Beats link:https://www.elastic.co/guide/en/beats/filebeat/current/keystore.html[documentation] for more details.
+
+
+[id="{p}-beat-set-beat-output"]
+=== Set Beat output
+
+If the `elasticsearchRef` element is specified, ECK populates the output section of the Beat config. ECK creates a user with appropriate roles and permissions and uses its credentials. If required, it also mounts the CA certificate in all Beat Pods, and recreates Pods when this certificate changes.
+
+Output can be set to any that's supported by a given Beat. To do that, remove the `elasticsearchRef` element from the specification and include an appropriate output configuration in the `config` or `configRef`.
```

Comment:
```suggestion
The `output` element can be set to any value that is supported by a given Beat. To use it, remove the `elasticsearchRef` element from the specification and include an appropriate `output` configuration in the `config` or `configRef` elements.
```

Reply:
I changed it a bit differently, I feel that referring to output part of Beat config in the same way as to CRD fields ("`output` element") might be confusing as there is no `output` in the CRD.